pandas/core/indexes/api.py

```diff
@@ -143,6 +150,13 @@ def _get_combined_index(
             index = index.sort_values()
         except TypeError:
             pass
+
+    # GH 29879
+    if copy:
+        # No need for deep copy. The underlying data of Indexes are
+        # considered immutable so they can be shared
+        index = index.copy(deep=False)
```

Comment:
you don't need the comment or deep=False which is the default.

Reply:
ok then. I wanted to be a bit more explicit, but sure if you think it's unnecessary.

Have cleaned it up.